src/gui/tray/ActivityItem.qml

```diff
@@ -24,36 +25,76 @@ MouseArea {
     ToolTip.visible: containsMouse && displayLocation !== ""
     ToolTip.delay: Qt.styleHints.mousePressAndHoldInterval
     ToolTip.text: qsTr("In %1").arg(displayLocation)
-        
+
     RowLayout {
         id: activityItem
-        
+
         readonly property variant links: model.links
-        
+
         readonly property int itemIndex: model.index
-        
+
         width: activityMouseArea.width
         height: Style.trayWindowHeaderHeight
         spacing: 0
-        
+
         Accessible.role: Accessible.ListItem
         Accessible.name: path !== "" ? qsTr("Open %1 locally").arg(displayPath)
                                      : message
-        Accessible.onPressAction: activityMouseArea.clicked()   
-        
-        Image {
-            id: activityIcon
+        Accessible.onPressAction: activityMouseArea.clicked()
+
+        Item {
             Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
-            Layout.leftMargin: 20
-            Layout.preferredWidth: shareButton.icon.width
-            Layout.preferredHeight: shareButton.icon.height
-            verticalAlignment: Qt.AlignCenter
-            cache: true
-            source: icon
-            sourceSize.height: 64
-            sourceSize.width: 64
+            Layout.leftMargin: 7
+            Layout.preferredWidth: shareButton.icon.width * 2
+            Layout.preferredHeight: shareButton.icon.height * 2
+
+            Image {
+                id: thumbnailImage
+                width: parent.width * 0.66
+                height: parent.height * 0.66
+                anchors.centerIn: parent
+                cache: true
+                source: model.thumbnail.source
+                visible: false
+                sourceSize.height: 64
+                sourceSize.width: 64
+            }
+
+            Item {
```

Comment:
is this `Item` really needed ?
cannot you directly instantiate the `Rectangle` ?

Reply:
Simplified with rectangle now